templates/problem/submit.html

```diff
@@ -107,6 +107,14 @@
                 $(window).resize(function () {
                     $('#ace_source').height(Math.max($(window).height() - 353, 100));
                 }).resize();
+
+                $('#problem_submit').submit(function (event) {
+                    if ($('#id_source').val().length > 65536) {
```

Comment:
Hilariously, this doesn't count it the same way as the server. In JavaScript, `'😭'.length == 2`, but we count `😭` as one character on the server.

Reply:
One solution we can use is to use the code described [here](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/String/length#Unicode) with using `[...str].length` instead of `str.length`, though I'm not sure if there are any implications.

From what I can remember, the purpose of making the check client-side is to prevent the site from timing out when submitting a few megabytes of source code (leaving the user clueless as to why their code couldn't be submitted). Another solution could be to increase the client bound to something like 262144 characters, and if the source code is between 65536 and 262144 characters, let the server handle the error message (which the server can handle reasonably quickly).

Reply:
I don't think this matters much, and `[...str].length` is not yet that compatible (we don't transpile JavaScript down, anyway).